### PR TITLE
Add section about formatting interpolated strings

### LINKED ISSUE #23306: F# style guide: How should long interpolated strings be formatted?
The F# code formatting guidelines do not currently specify how interpolated strings should be formatted.

To give an example, consider this string:

```f#
let serviceStorageConnection = $"DefaultEndpointsProtocol=https;AccountName=%s{serviceStorageAccount.Name};AccountKey=%s{serviceStorageAccountKey.Value}"
```

Fantomas currently (https://github.com/fsprojects/fantomas/issues/1511) formats this as:

```f#
let serviceStorageConnection =
    $"DefaultEndpointsProtocol=https;AccountName=%s{serviceStorageAccount.Name};AccountKey=%s{
                                                                                                  serviceStorageAccountKey.Value
    }"
```

Personally I don't find that particularly pretty, but long strings rarely are, and I don't really have a better suggestion right now. (Allowing deindenting interpolated expressions could help a lot here, though that belongs elsewhere. I won't bother with that right now.)

I can create a PR as long as someone can actually tell me how it should be formatted. I currently don't have any strong opinions. I want to note that the current way Fantomas formats it violates the "avoid name-sensitive alignment" clause, though without allowing deindenting interpolated expressions at the language level, the only ways out of that are 1) don't break (keep the entire interpolated string on a single line, or 2) (manually) break and concatenate the string. Personally I'm in favor of 1). Strings are the one thing where I don't bother keeping line widths.

/cc @cartermp @nojaf

### PR DESCRIPTION
**Pull request by cmeeren**
Fixes #23306

/cc @cartermp

### CODE REVIEW

**Comment by cmeeren** [comment]
Note: The point here is not to say that the string must necessarily be on the same line as the `let`. For example, for long string literals, Fantomas would do

```f#
let myString =
    "some very long string"
```

and not

```f#
let myString = "some very long string"
```

File: docs/fsharp/style-guide/formatting.md [hunk 1]
```diff
@@ -301,6 +301,15 @@ Inline comments should capitalize the first letter.
 let f x = x + 1 // Increment by one.
 ```
 
+Formatting interpolated strings
+-------------------------------
+
+Interpolated strings (and plain strings literals) can just be left on a single line, regardless of how long the line is.
```
Comment (nojaf) [inline docs/fsharp/style-guide/formatting.md:307]: How can `plain strings literals` be written any differently?

File: docs/fsharp/style-guide/formatting.md [hunk 2]
```diff
@@ -301,6 +301,15 @@ Inline comments should capitalize the first letter.
...
+
+```fsharp
+let serviceStorageConnection = $"DefaultEndpointsProtocol=https;AccountName=%s{serviceStorageAccount.Name};AccountKey=%s{serviceStorageAccountKey.Value}"
```
Comment (nojaf) [inline docs/fsharp/style-guide/formatting.md:310]: Put this on the next line, please.

File: docs/fsharp/style-guide/formatting.md (hunk 1 above)
Comment (cmeeren) [reply docs/fsharp/style-guide/formatting.md]: I guess that depends on why you want it written differently. I gave it a try now.

File: docs/fsharp/style-guide/formatting.md (hunk 2 above)
Comment (cmeeren) [reply docs/fsharp/style-guide/formatting.md]: Done.

**Comment by cartermp** [comment]
As noted here: https://github.com/dotnet/docs/issues/23306#issuecomment-801424146

We should probably also have a short bit about when you have multi-line interpolated expressions in an interpolated string. The gist of it should be:

1. Indent like the sample @nojaf gave
2. Use triple quotes (you likely will need to do that anyways)

**Comment by cmeeren** [comment]
>     2. Use triple quotes (you likely will need to do that anyways)

AFAIK you only need triple quotes if you have string literals inside the interpolated expression. If this is correct, do you still want me to add this point?

**Review by cartermp — APPROVED** [review]
Approved subject to markdownlint being happy :)

**Comment by cmeeren** [comment]
Markdownlint should now be happy. :)

**Comment by cmeeren** [comment]
Anything more to be done here, or is it ready to merge?

**Comment by cartermp** [comment]
This is good, thanks!

**Pull request merged**